include/RAJA/pattern/kernel/For.hpp

```diff
@@ -59,10 +59,10 @@ namespace internal
  * Assigns the loop index to offset ArgumentId
  *
  */
-template <camp::idx_t ArgumentId, typename Data, typename... EnclosedStmts>
-struct ForWrapper : public GenericWrapper<Data, EnclosedStmts...> {
+template <camp::idx_t ArgumentId, typename Data, typename Types, typename... EnclosedStmts>
+struct ForWrapper : public GenericWrapper<Data, Types, EnclosedStmts...> {
 
-  using Base = GenericWrapper<Data, EnclosedStmts...>;
+  using Base = GenericWrapper<Data, Types, EnclosedStmts...>;
```

Comment:
It feels a bit unfortunate that this needs to be passed through explicitly rather than being a part of the Data type.  Is there something that makes it better to keep it separate?

Reply:
It's type information that changes as we descend through the StatementExecutors.  I needed a way to change types within each For statement.  I didn't want to have to do some nasty casting of LoopData in order to accomplish this.  I thought this would be cleaner since it separates things into two concerns:  LoopData contains the runtime varying values, and LoopTypes contains the runtime "varying" types.

Reply:
Makes sense, thanks.